source/fe/fe_system.cc

```diff
@@ -1204,8 +1118,28 @@ compute_fill (const Mapping<dim,spacedim>                      &mapping,
                           fe_data.update_flags);
 
 
-  if (flags & (update_values | update_gradients))
+  if (flags & (update_values | update_gradients | update_hessians))
     {
+      if (fe_data.is_first_cell())
+        {
+          // Initialize the internal FE objects for the base elements
+          for (unsigned int base_no=0; base_no<this->n_base_elements(); ++base_no)
```

Comment:
I tried to just remove this block, but that's not enough. I think everything that happened in #1315 (commit 165be8) was overwritten.

The root cause for the confusion is that you didn't work on a branch. If you work on master, you can't easily pull the latest changes from upstream (because they would be applied *on top* of your changes when you want them to happen *below your changes*). The proper workflow is to work on a branch, and if you need to pull changes from upstream so you can rebase, you'd go back to master, pull changes there, go back to your branch, and rebase the branch on top of the latest changes you got onto master. Master in your repository should always correspond 1:1 to master in the upstream repository.

It's a common mistake and I made it too when I was new to git. It happens. You're quite obviously a good programmer -- you'll get used to working with git.

Can you see if you can manage to get your changes on top of what happened in #1315? I'd be happy to send it all through the testsuite in that case.